kratos/processes/apply_ray_casting_interface_recognition_process.h

```diff
@@ -0,0 +1,191 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Daniel Diez
+//
+
+#pragma once
+
+// System includes
+#include <string>
+#include <iostream>
+
+// External includes
+
+// Project includes
+#include "processes/apply_ray_casting_process.h"
+
+namespace Kratos
+{
+///@addtogroup Kratos Core
+///@{
+
+///@name Kratos Classes
+///@{
+
+/// Applies ray casting to distinguish the color (like in/out) of each node in modelpart
+/** This class is used to define the which nodes are inside, outside or on the interface
+ *  of certain volume described by its contour
+ */
+template<std::size_t TDim = 3>
+class KRATOS_API(KRATOS_CORE) ApplyRayCastingInterfaceRecognitionProcess : public ApplyRayCastingProcess<TDim>
+{
+
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of ApplyRayCastingInterfaceRecognitionProcess
+    KRATOS_CLASS_POINTER_DEFINITION(ApplyRayCastingInterfaceRecognitionProcess);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    using BaseType = ApplyRayCastingProcess<TDim>;
+
+    /**
+     * @brief Construct a new ApplyRayCastingProcess object using volume and skin model parts
+     * Constructor without user defined extra rays epsilon, used to
+     * generate the extra rays when voting is required for coloring
+     * @param rVolumePart model part containing the volume elements
+     * @param rSkinPart model part containing the skin to compute
+     * the distance to as conditions
+     */
+    ApplyRayCastingInterfaceRecognitionProcess(
+        ModelPart& rVolumePart,
+        ModelPart& rSkinPart,
+        Parameters ThisParameters = Parameters());
```

Comment:
Considering that we're adding a new feature I'd directly go for a `Model-Parameters` constructor.

Reply:
Agree, what about the constructor with `FindIntersectedGeometricalObjectsProcess`? https://github.com/KratosMultiphysics/Kratos/blob/81deb698f608f8f188cc4e9ea7dfd9464655336d/kratos/processes/apply_ray_casting_process.h#L105
I think this one is not needed either, at least in this derived classs

Reply:
It would make sense if you want to reuse the intersections from another process in here... Nonetheless, I guess this is not your target use case so I'd remove it. We can always add it upon demand.

Reply:
hmm well in that case I think it makes to leave it as it is now.